pkg/sql/plan/function/builtin/multi/substr.go

```diff
@@ -32,7 +34,12 @@ func castConstAsInt64(vec *vector.Vector, idx int64) int64 {
 	case types.T_uint32:
 		return int64(vector.GetValueAt[uint32](vec, idx))
 	case types.T_uint64:
-		return int64(vector.GetValueAt[uint64](vec, idx))
+		val := vector.GetValueAt[uint64](vec, idx)
```

Comment:
This is indeed weird.    Why not just throw an error?

Reply:
ok